### PR TITLE
Added settings object for update settings.

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

**Comment by pgdr** [comment]
seems a `plot_settings` test fails:

`263 - python.tests.ert.enkf.plot.plot_settings (Failed)`

**Comment by joakim-hove** [comment]
> seems a plot_settings test fails:

Yes - and that was actually more difficult than I thought to fix; working on it.

File: docs/user/keywords/index.rst [hunk 1]
```diff
@@ -1783,6 +1773,47 @@ The two keywords SETENV and UPDATE_PATH can be used to manipulate the Unix envir
...
+
+	Including outliers in the Smoother algorithm can dramatically
+	increase the coupling between the ensemble members. It is
+	therefore important to filter out these outlier data prior to
+	data assimilation. An observation, \textstyle d^o_i, will be
+	classified as an outlier if
+
+	::
+
+		|d^o_i - \bar{d}_i| > \mathrm{ENKF\_ALPHA} \left(s_{d_i} + \sigma_{d^o_i}\right),
```
Comment (pgdr) [inline docs/user/keywords/index.rst]: no need to change, but `\mathrm` can (and should) in most cases be simply `\text` or `\textnormal`.

`\mathrm` means that you want to go to roman font, which you should rarely (if ever) do.  `\text` simply switches to text mode.

File: libenkf/src/analysis_config.c [hunk 2]
```diff
@@ -38,6 +39,10 @@
 #include <ert/enkf/analysis_iter_config.h>
 
 
+#define UPDATE_OVERLAP_KEY      "OVERLAP_LIMIT"
```
Comment (pgdr) [inline libenkf/src/analysis_config.c]: Why is alpha removed as part of the name here but not in the API?

**Comment by joakim-hove** [comment]
> The only comment I have is that the config_settings should have been called a multityped_map or whatever and been placed in util.

Yes - that might be; it is quite related to config input though?

> The next time I need a map in ert, I'm not going to use config_settings_type.

Why? Because it is located in a unlogical place, or because you don't like this particular implementation?

**Comment by pgdr** [comment]
> Why? Because it is located in a unlogical place, or because you don't like this particular implementation?

Nono, I like the implementation!  It's just that I will look in `util` for something with `map` in its name.

**Comment by joakim-hove** [comment]
> Nono, I like the implementation! It's just that I will look in util for something with map in its name.

OK good - I understand; then I suggest we keep it here for now (it is reasonably tighly bound the config system afterall), and then we might extract parts of it to util later.

**Comment by pgdr** [comment]
Agreed.

**Pull request merged**